src/main/java/com/uber/cadence/internal/replay/HistoryHelper.java

```diff
@@ -82,14 +83,23 @@
       return decisionEvents;
     }
 
-    public HistoryEvent getDecisionEvent(long eventId) {
+    HistoryEvent getDecisionEvent(long eventId) {
       int index = (int) (eventId - nextDecisionEventId);
       if (index < 0 || index > decisionEvents.size()) {
         throw new IllegalArgumentException("No decision event found at eventId=" + eventId);
       }
       return decisionEvents.get(index);
     }
 
+    Optional<HistoryEvent> getOptionalDecisionEvent(long eventId) {
+      try {
+        HistoryEvent event = getDecisionEvent(eventId);
+        return Optional.of(event);
+      } catch (Exception e) {
+        return Optional.empty();
```

Comment:
Never use exceptions for flow control.
Either duplicate the code or implement getDecisionEvent in terms of getOptionalDecisionEvent.